### ISSUE TITLE
Using io.TextIOWrapper on S3File crashes on Python 2.7

### ISSUE THREAD

**Issue by dargueta** [dask/s3fs#130]
`S3File` cannot be fully used with [`io.TextIOWrapper`](https://docs.python.org/3/library/io.html#io.TextIOWrapper) on Python 2.7 because it doesn't provide all of the methods needed by `readline()`. This is a problem because files are always opened in binary mode, and if we need to access the file as text, this is the simplest way to do it without downloading the entire file.

Minimal example:

```py
from __future__ import print_function

import io

import moto
import s3fs

with moto.mock_s3():
    fs = s3fs.S3FileSystem()
    fs.mkdir('bucket')
    with fs.open('bucket/file.txt', 'wb') as fd:
        fd.write(u'\u00af\\_(\u30c4)_/\u00af'.encode('utf-16-le'))
    with fs.open('bucket/file.txt', 'rb') as fd:
        with io.TextIOWrapper(fd, 'utf-16-le') as stream:
            print(stream.readline())    # crashes here
```

The above code crashes with this:

```
AttributeError: 'S3File' object has no attribute 'read1'
```

**Comment by dargueta** [comment]
Fixed in #131

**Comment by dargueta** [comment]
Fixed in #131 .
